openpype/hosts/fusion/plugins/publish/validate_frame_range.py

```diff
@@ -0,0 +1,79 @@
+import pyblish.api
+
+from openpype.pipeline import (
+    PublishValidationError,
+    OptionalPyblishPluginMixin,
+)
+from openpype.pipeline.publish import RepairContextAction
+
+from openpype.pipeline.context_tools import get_current_project_asset
+from openpype.hosts.fusion.api.lib import (
+    set_asset_framerange,
+)
+
+
+class ValidateFrameRange(
+    pyblish.api.ContextPlugin,
+    OptionalPyblishPluginMixin,
+):
+    """Validate if the comp has the correct frame range"""
+
+    order = pyblish.api.ValidatorOrder
+    label = "Validate Frame Range"
+    families = ["render"]
+    hosts = ["fusion"]
+    optional = True
+    actions = [RepairContextAction]
+
+    def process(self, context):
+        if not self.is_active(context.data):
+            return
+
+        asset_doc = get_current_project_asset()
+        start = asset_doc["data"]["frameStart"]
+        end = asset_doc["data"]["frameEnd"]
+        handle_start = asset_doc["data"]["handleStart"]
+        handle_end = asset_doc["data"]["handleEnd"]
+
+        # Convert any potential none type to zero
+        handle_start = handle_start or 0
+        handle_end = handle_end or 0
+
+        # Calcualte in/out points
+        range_start = start - handle_start
+        range_end = end - handle_start
+
+        invalid = self.get_invalid(context, range_start, range_end)
+        if invalid:
+            raise PublishValidationError(
+                "The comp's frame range isn't correct"
+                "compared to the projects properties."
+                "\n\n{}".format("\n".join(invalid)),
+                title=self.label,
+            )
+
+    @classmethod
+    def get_invalid(cls, context, range_start, range_end):
+        invalid = []
+
+        if range_start != context.data["frameStartHandle"]:
```

Comment:
Technically frameStartHandle could match but frameStart and handleStart could mismatch e.g.

frameStart 100 and handleStart 10
frameStart 90 and handleStart 0

Both result in the same range start but they aren't the same. I feel like this validator should more explicitly check both the handles and frame ranges.

Reply:
The check is for the global start/end time and not the render start/end time as the render happens on the global range.
If the render range is different it doesn't matter as the render itself will use the global range

Reply:
I understand that it generates the same image frames, but the published instance will still have the frames and handles stored in the database which then still doesn't match the asset.